pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -427,6 +427,36 @@ func (fs *Decomposedfs) ListStorageSpaces(ctx context.Context, filter []*provide
 
 }
 
+func (fs *Decomposedfs) checkNodePermissions(ctx context.Context, requestedUserID string, unrestricted bool) bool {
+	authenticatedUserID := ctxpkg.ContextMustGetUser(ctx).GetId().GetOpaqueId()
+	canListAllSpaces := fs.canListAllSpaces(ctx)
+	switch {
+	case canListAllSpaces && requestedUserID == spaceTypeAny:
+		return true
+	case canListAllSpaces && authenticatedUserID != requestedUserID:
+		// as admin you have to be able to see other users spaces
+		return false
+	case canListAllSpaces, unrestricted:
+		// standard case, requesting user needs to check space permissions
+		return false
+	}
+	return true
```

Comment:
The func is unclear to me. But maybe that is just because of naming?
- What exactly is the function supposed to do? I mean what does it mean if this function returns `true`?
- The parameter `unrestricted` is unclear to me. What is unrestricted? And why does it result in a `false`?
- `requestedUserID == spaceTypeAny` makes no sense to me. How should a userID and a SpaceType match?

Reply:
the parameter `unrestricted` ins inherited from `ListStorageSpaces` above.

Reply:
if it returns true, then permissions on the nodes that are being red from the decomposedfs need to be checked, e.g. 
- user einstein wants to read spaces of user marie => does not have the permission => spaces won't be visible to that user.
- user einstein wants to read its own spaces => has permission => own spaces will be listed